src/main/java/fi/otavanopisto/kuntaapi/server/integrations/management/ManagementAnnouncementEntityUpdater.java

```diff
@@ -0,0 +1,183 @@
+  
+  @Inject
+  private ManagementApi managementApi;
+  
+  @Inject
+  private IdentifierController identifierController;
+
+  @Inject
+  private AnnouncementCache announcementCache;
+  
+  @Inject
+  private ModificationHashCache modificationHashCache;
+  
+  @Resource
+  private TimerService timerService;
+
+  private boolean stopped;
+  private List<AnnouncementIdUpdateRequest> queue;
+
+  @PostConstruct
+  public void init() {
+    queue = Collections.synchronizedList(new ArrayList<>());
+  }
+
+  @Override
+  public String getName() {
+    return "management-announcements";
+  }
+
+  @Override
+  public void startTimer() {
+    startTimer(TIMER_INTERVAL);
+  }
+
+  private void startTimer(int duration) {
+    stopped = false;
+    TimerConfig timerConfig = new TimerConfig();
+    timerConfig.setPersistent(false);
+    timerService.createSingleActionTimer(duration, timerConfig);
+  }
+
+  @Override
+  public void stopTimer() {
+    stopped = true;
+  }
+  
+  @Asynchronous
+  public void onAnnouncementIdUpdateRequest(@Observes AnnouncementIdUpdateRequest event) {
+    if (!stopped) {
+      AnnouncementId announcementId = event.getId();
+      
+      if (!StringUtils.equals(announcementId.getSource(), ManagementConsts.IDENTIFIER_NAME)) {
+        return;
+      }
+      
+      if (event.isPriority()) {
+        queue.remove(event);
+        queue.add(0, event);
+      } else {
+        if (!queue.contains(event)) {
+          queue.add(event);
+        }
+      }
+    }
+  }
+  
+  @Asynchronous
+  public void onAnnouncementIdRemoveRequest(@Observes AnnouncementIdRemoveRequest event) {
+    if (!stopped) {
+      AnnouncementId announcementId = event.getId();
+      
+      if (!StringUtils.equals(announcementId.getSource(), ManagementConsts.IDENTIFIER_NAME)) {
+        return;
+      }
+      
+      deleteAnnouncement(event, announcementId);
+    }
+  }
+  
+  @Timeout
+  public void timeout(Timer timer) {
+    if (!stopped) {
+      if (!queue.isEmpty()) {
+        AnnouncementIdUpdateRequest updateRequest = queue.remove(0);
+        updateManagementAnnouncement(updateRequest.getOrganizationId(), updateRequest.getId());
+      }
+
+      startTimer(SystemUtils.inTestMode() ? 1000 : TIMER_INTERVAL);
+    }
+  }
+  
+  private void updateManagementAnnouncement(OrganizationId organizationId, AnnouncementId announcementId) {
+    DefaultApi api = managementApi.getApi(organizationId);
+    
+    ApiResponse<Announcement> response = api.wpV2AnnouncementIdGet(announcementId.getId(), null);
+    if (response.isOk()) {
+      updateManagementAnnouncement(organizationId, response.getResponse());
+    } else {
+      logger.warning(String.format("Find organization %s announcement %s failed on [%d] %s", organizationId.getId(), announcementId.toString(), response.getStatus(), response.getMessage()));
+    }
+  }
+  
+  private void updateManagementAnnouncement(OrganizationId organizationId, Announcement managementAnnouncement) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "updateManagementAnnouncement" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)